python/cugraph-service/server/pyproject.toml

```diff
@@ -45,7 +45,7 @@ dynamic = ["entry-points"]
 test = [
     "aiohttp",
     "fsspec[http]>=0.6.0",
-    "networkx>=2.5.1",
+    "networkx>=3.0",
```

Comment:
Is it really necessary to update nx versions in other packages? It's probably simplest to keep dependency versions between packages in-sync, but cugraph-nx may at times want to depend on pretty new versions of networkx.

Reply:
> Is it really necessary to update nx versions in other packages?

possibly not, `dependencies.yaml` makes it easy to align everything so I ended up doing that, but I'd be in favor of having all cugraph components always work with/depend on the latest stable Nx release though anyway.  Especially since (I think) we'll be migrating use cases that require cugraph to import networkx to `cugraph-nx` hopefully soon anyway.

Do you think aligning all cugraph components (including cugraph-nx) on the latest stable networkx will be a problem?  If so, I can make it so `cugraph-nx` depends on a different version of networkx than all other cugraph packages.

Reply:
well...now I'm thinking about it more and perhaps we don't want to force Nx upgrades on people that don't want them, as long as we're compatible with the older versions.  `cugraph-nx` is a different case though in that it may require newer versions.  For that reason, we could have the other cugraph packages specify a wider range, eg. `>=2.5.1`.

However, keeping them all aligned could simply things such as development envs (which will probably just have to have a newer version to accommodate both cugraph* and cugraph-nx development), and CI runs which will benefit from scripts that only need to worry about a single networkx install.

I think I'm leaning towards aligning everything on the latest stable networkx, but could be convinced otherwise.  Thoughts?

Reply:
I think it's a good idea for all cugraph packages to be _compatible_ with the latest stable release of networkx.

For now, I expect cugraph-nx to be compatible with networkx versions `>=3.0`. However, I don't think it's worth much development effort for cugraph-nx to remain backwards compatible, and it's possible--even likely--that we'll want to depend on very recent versions in the future.

So, from my perspective, I want the freedom to be able to choose the networkx versions supported by cugraph-nx, and I don't think this should dictate what other cugraph packages support. What is the cost of doing it this way?

Reply:
I think from the perspective of allowing different recipes to have different networkx version specs, we can do that pretty easily (we can have everything use the `networkx` anchor in `dependencies.yaml` except `cugraph-nx` which will specify its own version spec).  

However, it gets harder for our generated conda dev envs and CI scripts, which will likely need to settle on a single networkx version to keep from getting too complicated.  From that perspective, the `cugraph-nx` networkx minimum version will dictate what other cugraph packages have to support.  I'm fine with that though since we want cugraph* to work with the latest anyway, plus the amount of cugraph functionality that uses networkx is decreasing over time so my guess is we won't hit many (or any) incompatibilities anyway (famous last words, I know).

So I'm interpreting this as let's allow the `cugraph-nx` recipe to have a higher minimum version than the rest of cugraph, but test and develop all of cugraph with the version required by `cugraph-nx`.  If we run into incompatibilities, we update the cugraph code, remove the dependency, or look into different dev/test setups at that time.  Does that sound reasonable?

Reply:
Yeah, that sounds great to me. Let's plan to keep `cugraph-nx` working with the normal dev environment. This may limit how "bleeding edge" `cugraph-nx` can be, but the alternative seems worse.

Reply:
This might be resolved in the latest commit, but I want to see if it's okay that `rapids-dependency-file-generator` is [adding both networkx versions in the dev env file](https://github.com/rapidsai/cugraph/pull/3793#discussion_r1300290379).

Reply:
This won't cause any issues for dfg itself. dfg does deduplicate dependencies, but it takes the entire dependency spec as a unit so the same package with different constraints will be treated as two different dependencies. Having two compatible constraints shouldn't cause any problems for pip or conda solvers though, so I don't see any issues with that.